libvmi/core.c

```diff
@@ -367,9 +369,11 @@ set_id_and_name(
     if (VMI_SUCCESS == driver_get_name_from_id(vmi, id, &tmp_name)) {
         dbprint(VMI_DEBUG_CORE, "--got name from id (%"PRIu64" --> %s)\n", id, tmp_name);
         vmi->image_type = strndup(tmp_name, 100);
```

Comment:
driver_get_name_from_id returns VMI_FAILURE if it failed, so if we are in here, we know that tmp_name is not NULL.